.jenkinsci/helpers/hash.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+
+import hashlib
+import argparse
+import os
+
+WINDOWS_LINE_ENDING = b'\r\n'
+UNIX_LINE_ENDING = b'\n'
+
+
+def md5_update_from_dir(directory, hash):
+    assert os.path.isdir(directory)
+    for root, dirnames, filenames in os.walk(directory):
+        for file in sorted(filenames, key=lambda p: str(p).lower()):
+            # If you need include file name to hash uncomment this
+            #hash.update(file.encode())
```

Comment:
```suggestion
            hash.update(file.encode())
```

Reply:
do we need it? the hash will be same if we change the register of filename, but if we rename the sort order will change and we get new hash. I am afraid that name can contain platform depended symbols, (ex: path delimiter  linux: '/'  windows: '\')

Reply:
okay, seems fine